Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -115,95 +114,91 @@ class TransactionHistoryViewController: UIViewController {
         if self.btnFilter.selectedItemIndex >= 0 && self.filtertitle.count > self.btnFilter.selectedItemIndex {
             currentFilterItem = self.filtertitle[self.btnFilter.selectedItemIndex]
         }
-        self.updateFilterDropdownItems()
         self.applyTxFilter(currentFilter: currentFilterItem)
     }
     
     func applyTxFilter(currentFilter: Int) {
+        var filterCount: Int? = 0
         switch currentFilter {
-        case 1:
-            // TODO: Remove after next dcrlibwallet update
-            self.Filtercontent = self.mainContens.filter{$0.Direction == 0 && $0.Type == GlobalConstants.Strings.REGULAR}
-            self.btnFilter.setTitle(LocalizedStrings.sent.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        case 2:
-            // TODO: Remove after next dcrlibwallet update
-            self.Filtercontent = self.mainContens.filter{$0.Direction == 1 && $0.Type == GlobalConstants.Strings.REGULAR}
-            self.btnFilter.setTitle(LocalizedStrings.received.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
+            case 1:
+                filterCount = transactions.filter{$0.Direction == 0 && $0.Type == GlobalConstants.Strings.REGULAR}.count
+                self.btnFilter.setTitle(LocalizedStrings.sent.appending("(\(filterCount!))"), for: .normal)
```

Comment:
Why does `applyTxFilter` update the text on the filter button? Shouldn't this function perform a tx filter and display the txs matching the `currentFilter`?

Reply:
this behavior has been corrected in